### PR TITLE
chore(release): prepare v3.6.0

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary

Prepares the v3.6.0 release on top of v3.5.0: version bump plus regression coverage for the opt-in active-input layer (mouse-wheel capture and gamepad passthrough suppression) and matching docs.

## Summary by CodeRabbit

* **Version**
  * Project version bumped to 3.6.0

* **Tests**
  * Expanded input interception test coverage with live integration tests for window-procedure and XInput hook functionality, including edge cases like wheel-message handling and hot-reload scenarios.

* **Documentation**
  * Updated test coverage guide to reflect newly automated input interception testing; clarified which scenarios remain manual validation.

### CODE REVIEW

File: tests/test_input_intercept.cpp [hunk 1]
```diff
@@ -166,3 +174,381 @@ TEST(GamepadSuppressTest, RepressDuringGraceReHolds)
...
+    // After the self-heal a freshly created window can be subclassed again.
+    hwnd_ = make_test_window();
+    if (hwnd_ == nullptr)
+    {
+        GTEST_SKIP() << "no window station available to recreate a window";
+    }
+    if (install_on_our_window())
+    {
+        EXPECT_TRUE(wndproc_installed());
+    }
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply tests/test_input_intercept.cpp]: @coderabbitai

Fixed in 699f28d. The re-subclass tail now mirrors the guard at thetop of the test: if `install_on_our_window()` returns false it `GTEST_SKIP()`s with a message, otherwise `EXPECT_TRUE(wndproc_installed())` always runs. No more silent green on the re-subclass path.
